Integrate new icons

adamwight opened the pull request:
Provide some backwards-compatibility, and rebuild the maki-7 icons at the legacy sizes which are expected by the rest of our software stack.

Includes a major rewrite of alphanumeric rendering.

adamwight committed:
Map icon names to newer names

Background from thiemowmde:

"chemist" is heavily ambiguous. Could be a science lab on a
campus. Could be an industrial site. Could be an alternative for a
pharmacy.  A quick image search leaves the impression that no country
uses a test glass to represent a pharmacy. The Maki 2 announcement
mentions Noun Project as inspiration. Searching that for "chemist"
and "pharmacy" doesn't make the situation much clearer.  The patch
that added the icon associates it with many different things.
dewikivoyage translates it as "Drogerie".  A Maki 2 issue from 2016
associates it with "pharmacy".

There are many possible replacements for "minefield", e.g.  "barrier"
or "caution". "danger" felt most appropriate.

"london-underground", "rail-above" and "rail-underground" become
"rail". I couldn't find a better alternative.

adamwight committed:
Drop xtend dependency

This module was deprecated by es6 syntax.

adamwight committed:
Rerender symbols, at legacy sizes

The old (maki 0.5.0) icons were rendered with a large margin which is
wasted space in a bitmap.  Instead, this patch renders the full
15x15 (units) new icons in smaller boxes with no margin.  Exact sizes
were chosen such that new icons are roughly congruent to the old ones.

There is a slight savings in the number of bits for each image.

Includes a few more symbols.

adamwight committed:
Verbose filename during render

adamwight committed:
Bump major version to 4 to reflect Maki update

adamwight committed:
Cache the new image sizes

adamwight committed:
Generate alphanumeric SVG source dynamically

This still builds static PNGs to serve, but the source step is done
by a script.

No longer paints a stroke around glyphs, to match the new Maki style
guidelines.

adamwight committed:
Remove deprecated alphanum master doc

adamwight committed:
Point to the @wikimedia repo

thiemowmde commented on index.js:
```diff
@@ -17,6 +17,25 @@ var offsets = {
     sizes = { s: 12, m: 18, l: 24 },
     makiRenders = __dirname + '/renders/';
 
+var aliases = {
```
A few last minute changes are missing here.
* chemist → blank
* london-underground → rail-metro
* rail-underground → rail-metro

See https://phabricator.wikimedia.org/T300040#7737994 for reference.

thiemowmde commented on package.json:
```diff
@@ -35,5 +35,5 @@
   "bugs": {
     "url": "https://github.com/mapbox/makizushi/issues"
```
If this is now focused on our fork instead of the original, should we use https://phabricator.wikimedia.org/tag/kartotherian/ here?

thiemowmde commented on index.js:
```diff
@@ -1,7 +1,6 @@
 var fs = require('fs'),
     path = require('path'),
```
Unrelated, but it looks like the `path` dependency here was never used.

thiemowmde reviewed (commented):
I'm afraid I'm still very lost.
* Where is the code that renders the little icon .png files on top of the "pin" bases? I can find the empty pins, but no code.
* ~~Where is the code that renders the numbers 1–99 and the letters A–Z? I can find a file markers-src/alphanum/build-retina.sh, but it doesn't look like it can work any more. It still references the deleted alphanum.svg.~~

adamwight commented:
Force-pushed to squash the alphanumeric work.  No longer stroke around alphanum glyphs.

adamwight commented:
>     * Where is the code that renders the little icon .png files on top of the "pin" bases? I can find the empty pins, but no code.

The script to render symbols from svg to png is in the top-level directory, render.sh

Code to bit-blit a symbol png onto the marker and mask is in index.js

adamwight committed:
Clean-up; minor updates to package.json

adamwight committed:
Update how we alias deprecated labels

adamwight committed:
Support aliasing to a blank pin

This was going to be used for "chemist", mapping it to the blank pin
by giving it a target icon of "", but we might not do so.

Now the code is at least organized to make such a thing possible in
the future.

adamwight committed:
Update changelog

adamwight committed:
Flip symbol fill from white to black when relative luminance crosses a threshold

This makes the symbol visible on light backgrounds.

adamwight committed:
Fix relative luminance calculation

Was missing a color space conversion.  The resulting formula has been
checked against other references.

adamwight committed:
Use Base10 as the dark symbol color

Calculate the luminance threshold dynamically, improving code
maintainability in case we need to change the color again.

adamwight committed:
Move source to lib directory

adamwight committed:
Extract color handling into its own module

adamwight committed:
Unit tests for color methods

adamwight committed:
Cache all symbols

Drops the code path for reading files at runtime.

adamwight committed:
Map alternate park old name to new name

Note that the tree type is flipped between versions, but we'll trust
that the map author's intention is usually "recognizable icon for
park" rather than a specific tree species.

adamwight committed:
Tweak contrast threshold to value found using the APCA algorithm

This switches to dark fills only once the background is much lighter
than before, roughly at #a7a7a7.
